shared/common-adapters/header-hoc.desktop.js

```diff
@@ -8,31 +8,33 @@ import {globalStyles, globalColors, globalMargins} from '../styles'
 
 import type {Props} from './header-hoc'
 
-function HeaderHoc<P>(WrappedComponent: React.ComponentType<P>) {
-  return ({onBack, onCancel, headerStyle, title, theme = 'light', ...restProps}: Props & P) => (
-    <Box style={_containerStyle}>
-      <Box style={{..._headerStyle, ..._headerStyleThemed[theme], ...headerStyle}}>
-        {onBack &&
-          <BackButton
-            key="back"
-            onClick={onBack}
-            style={{..._backButtonIconStyle, ..._backButtonIconStyleThemed[theme]}}
-          />}
-        {onCancel &&
-          <Icon
-            style={{..._styleClose, ..._styleCloseThemed[theme]}}
-            type="iconfont-close"
-            onClick={onCancel}
-          />}
-        {title &&
-          <Box style={_titleStyle}>
-            <Text type="Header">{title}</Text>
-          </Box>}
+function HeaderHoc<P: {}>(WrappedComponent: React.ComponentType<P>) {
+  return (props: P & Props) => {
+    const {onBack, onCancel, headerStyle, title, theme = 'light'} = props
+    return (
+      <Box style={_containerStyle}>
+        <Box style={{..._headerStyle, ..._headerStyleThemed[theme], ...headerStyle}}>
+          {onBack &&
+            <BackButton
+              key="back"
+              onClick={onBack}
+              style={{..._backButtonIconStyle, ..._backButtonIconStyleThemed[theme]}}
+            />}
+          {onCancel &&
+            <Icon
+              style={{..._styleClose, ..._styleCloseThemed[theme]}}
+              type="iconfont-close"
+              onClick={onCancel}
+            />}
+          {title &&
+            <Box style={_titleStyle}>
+              <Text type="Header">{title}</Text>
+            </Box>}
+        </Box>
+        <WrappedComponent {...(props: P)} theme={theme} onBack={onBack} onCancel={onCancel} />
```

Comment:
I think it's more correct to pass in `props` instead of `restProps`, since theoretically P and Props can share keys.